### ISSUE TITLE
Journal Mode

### ISSUE THREAD

**Issue by rolandjohann** [BoostIO/Boostnote#1072]
As said at #1071 this can be a perfect solution for a journaling app, but a feature is missing:

# Journal Mode

This will be a separate view where one can create a journal entry with meta:
- tags
- heading/title
- content

Based on the journal entries the notes will be built. The tags define the notes they belong to, heading defines subheadlines. If the same headings are defined multiple times, the content get appended to that section.

It's worth to think about this and tweak tagging/heading to get a better UX - for example:
A note is likely to have more than two levels of headlines. This structure needs to be mapped to the concept depicted ahead: tagging and heading of the journal entry. Maybe this can be done by hierarchical tags. These can be of special syntax `some tag::and it's headline::second headline`.

At the end of the day it should be possible to throw a journal entry into the app specifying its content, some tags and a heading/title and are built to a collection of notes acting as wiki.

Additional meta information will be stored automatically, like date time, maybe it should be possible to define further ones like `project`. I can imagine that meta like `project` can be mapped into tags and become a separate note for each project...

**Comment by GaryFurash** [comment]
LOVE IT!

**Comment by markerikson** [comment]
Yep, would love to see some kind of journaling support.  Playing around with Boostnote some for the first time, it looks like I could sorta simulate things by giving notes names/first lines like "2017-12-03", and maybe creating categories per month or something.

I've been using https://rednotebook.sourceforge.io/ .  I really like its date-based UI and behavior, but the lack of Markdown support is driving me nuts.  Anything that could help mimic some of that capability would be great.

**Comment by ttdtrang** [comment]
I'm trying to implement some kind of Journal mode. It works something like this
* To create a journal entry, you must be in a **JOURNAL** folder. This way specific requirements of a journal entry is satisfied without interfering with other note collections.
* A journal entry can only be created once for a given date.
* A journal entry is created by clicking on the target date on a calendar.

What's hard to decide is whether to use the automatically recorded creation date as a journal date or to add another date field to record the journal entry date. For example, you are on Monday, Feb 26, but want to write the journal entry for a day in the past, Sat Feb 24, the journal mode should give you a new journal entry for Sat Feb 24, but will annotate it with creation date at Feb 26.

Right now I'm leaning toward keeping these dates separate. Please let me know if you have any thoughts about that.

**Comment by rolandjohann** [comment]
@ttdtrang the question in mind is if we mean the same with our journaling feature. It seems that you would like to have a daily journal.

Unfortunately my problem is more complex as I want to achieve that I can throw notes about certain topics with some tags/metadata into the app and be able to compile articles/documents afterwards based on tags and timestamp. With this we can build a great way to keep journals over a day, as the need for keeping notes arises.

My main problem with existing solutions is that I keep posting notes to different logs multiple times because there is no tagging at plain files ;)

**Comment by Flexo013** [comment]
I'm not quite sure if such a complex journal concept could work in Boostnote.

@rolandjohann If you had to narrow it down to simple features, that could be introduced one after another, then what steps would those be?

**Comment by Thyrst** [comment]
I like what @ttdtrang suggested. 


> What's hard to decide is whether to use the automatically recorded creation date as a journal date or to add another date field to record the journal entry date. For example, you are on Monday, Feb 26, but want to write the journal entry for a day in the past, Sat Feb 24, the journal mode should give you a new journal entry for Sat Feb 24, but will annotate it with creation date at Feb 26.
> 
> Right now I'm leaning toward keeping these dates separate. Please let me know if you have any thoughts about that.

Look how Zim wiki handles this. It is separated. However you can just change the creation date if you like to. This way it make sense to me.
